docs/configure.asciidoc

```diff
@@ -0,0 +1,326 @@
+////
+Goal of this doc:
+Provide a complete reference of all available configuration options and where/how they can be set. (Any Elastic-specific configuration options are listed directly. General OpenTelemetry configuration options are linked.)
+
+Assumptions we're comfortable making about the reader:
+* They are familiar with Elastic
+* They are familiar with OpenTelemetry
+////
+
+[[configure]]
+== Configure
+
+:language: .NET
+:language_lc: dotnet
+:distro_name: Elastic Distribution for OpenTelemetry {language}
+
+include::release-status.asciidoc[]
+
+The {distro_name} ("the distro") offers a lot of flexibility...
+
+////
+How do users set configuration options?
```

Comment:
Either through environment variables or `IConfiguration` the latter is a well known configuration method for .NET apps.

The source for which can be files/databases etc whatever the user choses. The example in the document uses a settings json file which is the most common.